lib/ansible/modules/cloud/amazon/ec2_lc.py

```diff
@@ -225,125 +325,172 @@ def create_launch_config(connection, module):
+
+    launch_config['LaunchConfigurationName'] = name
+
+    if security_groups is not None:
+        launch_config['SecurityGroups'] = security_groups
+
+    if classic_link_vpc_id is not None:
+        launch_config['ClassicLinkVPCId'] = classic_link_vpc_id
+
+    if instance_monitoring:
+        launch_config['InstanceMonitoring'] = {
+            'Enabled': advanced_instance_monitoring}
+
+    if classic_link_vpc_security_groups is not None:
+        launch_config['ClassicLinkVPCSecurityGroups'] = (
+            classic_link_vpc_security_groups)
+
+    if bdm is not None:
+        launch_config['BlockDeviceMappings'] = [bdm]
+
+    if instance_profile_name is not None:
+        launch_config['IamInstanceProfile'] = instance_profile_name
+
+    if assign_public_ip is not None:
+        launch_config['AssociatePublicIpAddress'] = assign_public_ip
+
+    if user_data is not None:
+        launch_config['UserData'] = user_data
+
+    if ebs_optimized is not None:
+        launch_config['EbsOptimized'] = ebs_optimized
+
+    if len(launch_configs) == 0:
         try:
-            connection.create_launch_configuration(lc)
-            launch_configs = connection.get_all_launch_configurations(names=[name])
+            connection.create_launch_configuration(**launch_config)
+            launch_configs = connection.describe_launch_configurations(
+                LaunchConfigurationNames=[name]).get('LaunchConfigurations')
             changed = True
-        except BotoServerError as e:
+            if len(launch_configs) > 0:
+                launch_config = launch_configs[0]
+            else:
+                module.fail_json(msg="error creating launch configuration. " +
+                                 "no launch configuration returned from boto3" +
+                                 " after create.")
+        except botocore.exceptions.ClientError as e:
             module.fail_json(msg=str(e))
 
-    result = dict(
-                 ((a[0], a[1]) for a in vars(launch_configs[0]).items()
-                  if a[0] not in ('connection', 'created_time', 'instance_monitoring', 'block_device_mappings'))
-        )
-    result['created_time'] = str(launch_configs[0].created_time)
-    # Looking at boto's launchconfig.py, it looks like this could be a boolean
-    # value or an object with an enabled attribute.  The enabled attribute
-    # could be a boolean or a string representation of a boolean.  Since
-    # I can't test all permutations myself to see if my reading of the code is
-    # correct, have to code this *very* defensively
-    if launch_configs[0].instance_monitoring is True:
-        result['instance_monitoring'] = True
+    result = (dict((k, v) for k, v in launch_config.items()
+              if k not in ['connection', 'created_time', 'instance_monitoring',
+              'block_device_mappings']))
+
+    result['CreatedTime'] = str(launch_config.get('CreatedTime'))
+
+    if launch_config.get('InstanceMonitoring') is True:
```

Comment:
Is this not returned by describe_launch_configurations unless it's true? Maybe rather than handle True and False and no value separately, can you just use the logic in the else statement below?